odbc/src/api/functions.rs

```diff
@@ -2761,9 +2804,49 @@ unsafe fn set_connect_attrw_helper(
     let sql_return = {
         let conn = must_be_valid!((*conn_handle).as_connection());
 
+        let has_connection = conn.has_mongo_connection();
+
+        // Attributes could be set before or after the mongo_connection is established, so we
+        // have to check and write to the proper place. Keep in mind the conn.attributes Option<RwLock<ConnectionAttributes>> is TAKEN
+        // when a mongo_connection is established, so writing to both isn't safe
```

Comment:
Dead comments that will be removed. We discussed offline that we won't bother validating if the dm is passing attributes to us when it should or shouldn't.